src/sidebar/session.js

```diff
@@ -208,10 +208,14 @@ function session($http, $resource, $rootScope, annotationUI, auth,
       auth.clearCache();
     }).catch(function (err) {
       flash.error('Log out failed');
-      throw err;
+      analytics.track(analytics.events.LOGOUT_FAILURE);
+      return $q.reject(new Error(err));
```

Comment:
_Note to self:_

It wasn't obvious to me here why the change was required, since throwing an exception in a `catch` handler should be the same as returning a rejected Promise. However, Angular's `$q` service contains this code:

```js
try {
  if (isFunction(fn)) {
    deferred.resolve(fn(state.value));
  } else if (state.status === 1) {
    deferred.resolve(state.value);
  } else {
    deferred.reject(state.value);
  }
} catch (e) {
  deferred.reject(e);
  exceptionHandler(e);
}
```

In the event that a Promise callback (`fn` above) throws an exception, `$q` invokes Angular's `exceptionHandler` service which causes the test to fail 🤦‍♂️ . This doesn't happen when returning a rejected Promise.

So this change is fine, it just took me some time to understand.